RecoLocalCalo/HcalRecAlgos/src/HBHENegativeFlag.cc

```diff
@@ -1,193 +1,28 @@
-#include "CondFormats/HcalObjects/interface/OOTPileupCorrData.h"
-
-#include "Geometry/CaloTopology/interface/HcalTopology.h"
-#include "Geometry/Records/interface/IdealGeometryRecord.h"
-//---------------------------------------------------------------------------
-HBHENegativeFlagSetter::HBHENegativeFlagSetter()
-{
-   //
-   // Argumentless constructor; should not be used
-   // 
-   // If arguments not properly specified for the constructor, I don't think
-   // we'd trust the flagging algorithm.
-   // Set the minimum charge threshold large enough so that nothing will be flagged.
-   // 
-
-   mMinimumChargeThreshold = 99999999;
-   mTS4TS5ChargeThreshold = 99999999;
-   mFirst = 4;
-   mLast = 6;
-   mBunchCrossingInfo = NULL;
-   mLengthBunchCrossingInfo = 0;
-}
-//---------------------------------------------------------------------------
-HBHENegativeFlagSetter::HBHENegativeFlagSetter(double minimumChargeThreshold,
-   double tS4TS5ChargeThreshold, int first, int last,
-   std::vector<double> threshold, std::vector<double> cut)
-{
-   //
-   // The constructor that should be used
-   //
-   // Copies various thresholds and limits and parameters to the class for future use.
-   //
-
-   mMinimumChargeThreshold = minimumChargeThreshold;
-   mTS4TS5ChargeThreshold = tS4TS5ChargeThreshold;
-   mFirst = first;
-   mLast = last;
-
-   if (mFirst < 2 || mLast < mFirst)
-       throw cms::Exception("Invalid mFirst, mLast specification");
-
-   mBunchCrossingInfo = NULL;
-   mLengthBunchCrossingInfo = 0;
-
-   for(int i = 0; i < (int)std::min(threshold.size(), cut.size()); i++)
-      mCut.push_back(std::pair<double, double>(threshold[i], cut[i]));
-   std::sort(mCut.begin(), mCut.end());
-}
-//---------------------------------------------------------------------------
-HBHENegativeFlagSetter::~HBHENegativeFlagSetter()
+void HBHENegativeFlagSetter::setPulseShapeFlags(
+    HBHERecHit &hbhe, 
+    const HBHEDataFrame &digi,
+    const HcalCoder &coder, 
+    const HcalCalibrations &calib)
 {
-   // Dummy destructor - there's nothing to destruct by hand here
-}
-//---------------------------------------------------------------------------
-void HBHENegativeFlagSetter::setPulseShapeFlags(HBHERecHit &hbhe, 
-   const HBHEDataFrame &digi,
-   const HcalCoder &coder, 
-   const HcalCalibrations &calib)
-{
-   //
-   // Decide if a digi/pulse is good or bad using negative energy discriminants
-   //
-   // SetPulseShapeFlags determines the total charge in the digi.
-   // If the charge is above the minimum threshold, the code then
-   // runs the flag-setting algorithms to determine whether the
-   // flags should be set.
-   //
-
-   const OOTPileupCorrData* corrObj = dynamic_cast<OOTPileupCorrData*>(hbhePileupCorr_.get());
-   if (corrObj)
+   if (filter_)
    {
       CaloSamples cs;
       coder.adc2fC(digi,cs);
       const int nRead = cs.size();
 
-      double ts[CaloSamples::MAXSAMPLES] {};
+      double ts[CaloSamples::MAXSAMPLES];
```

Comment:
Hi Chris and all

boost seems like an overkill for this pretty much C code.

At this point in discussion,  I'd rather stay with the current code.
Most developer cases that I wanted to cover here by asking for
initialization was the quick copy-paste-and-edit  (most of what we have in
some development areas). Copy paste of code with boost containers will have
more issues down the road.

Slava
On May 2, 2015 8:16 AM, "Chris Jones" <notifications@github.com> wrote:

> In RecoLocalCalo/HcalRecAlgos/src/HBHENegativeFlag.cc
> <https://github.com/cms-sw/cmssw/pull/8923#discussion_r29547370>:
>
> >     {
> >        CaloSamples cs;
> >        coder.adc2fC(digi,cs);
> >        const int nRead = cs.size();
> >
> > -      double ts[CaloSamples::MAXSAMPLES] {};
> > +      double ts[CaloSamples::MAXSAMPLES];
>
> I agree with slava that using a 'bare' array to represent an arbitrary
> sized container is likely to lead to maintenance problems down the road. I
> therefore suggest using a boost::static_vector<double,
> CaloSamples::MAXSAMPLES> to encapsulate the array and the size. See
>
> http://www.boost.org/doc/libs/1_54_0/doc/html/boost/container/static_vector.html
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/8923/files#r29547370>.
>